pilot/pkg/proxy/envoy/v2/discovery_test.go

```diff
@@ -285,9 +292,14 @@ func TestDebounce(t *testing.T) {
 
 			close(stopCh)
 			wg.Wait()
-
-			if partialPushes != tt.expectedPartial || fullPushes != tt.expectedFull {
-				t.Fatalf("Got %v full and %v partial, expected %v full and %v partial", fullPushes, partialPushes, tt.expectedFull, tt.expectedPartial)
+			err := retry.UntilSuccess(func() error {
```

Comment:
@howardjohn would it help if we embedded the wait logic (i.e. `UntilSuccess`) in each test directly so that they can perform intermediate checks rather than sleeping?

Reply:
Sent an update using this pattern, let me know what you think.